src/jsx/components/Student/Students.jsx

```diff
@@ -41,8 +43,8 @@ function Students() {
   };
 
   useEffect(() => {
-    getAll({ page: currentPage, order: sortOrder });
-  }, [currentPage, sortOrder]);
+    getAll({ q: query, page: currentPage, order: sortOrder });
```

Comment:
aquí podríamos evitar mandar el parámetro q con la cadena vacía. 
```suggestion
let params = { page: currentPage, order: sortOrder }

if (query.length) {
   params = {...params, q: query }
} 
getAll(params);
```